src/utils.ts

```diff
@@ -215,21 +219,64 @@ export const setCursor = (canvas: HTMLCanvasElement | null, cursor: string) => {
   }
 };
 
+let eraserCanvasCache: any;
+export const setEraserCursor = (
+  canvas: HTMLCanvasElement | null,
+  theme: AppState["theme"],
+) => {
+  const drawCanvas = () => {
+    const cursorImageSizePx = 20;
+    eraserCanvasCache = document.createElement("canvas");
+    eraserCanvasCache.height = cursorImageSizePx;
+    eraserCanvasCache.width = cursorImageSizePx;
+
+    const context = eraserCanvasCache.getContext("2d")!;
+    if (theme === THEME.DARK) {
+      context.strokeStyle = oc.white;
+    } else {
+      context.strokeStyle = oc.black;
+    }
+    context.beginPath();
+    context.arc(
+      eraserCanvasCache.width / 2,
+      eraserCanvasCache.height / 2,
+      5,
+      0,
+      2 * Math.PI,
+    );
+    context.stroke();
+  };
+  if (!eraserCanvasCache) {
+    drawCanvas();
+  } else {
+    const context = eraserCanvasCache.getContext("2d")!;
+    const strokeStyle = context.strokeStyle;
+    if (theme === THEME.LIGHT && strokeStyle !== oc.black) {
+      drawCanvas();
+    } else if (theme === THEME.DARK && strokeStyle !== oc.white) {
+      drawCanvas();
+    }
+  }
+
+  const previewDataURL = eraserCanvasCache.toDataURL(MIME_TYPES.svg) as DataURL;
+
+  setCursor(canvas, `url(${previewDataURL}) 4 4, auto`);
```

Comment:
didn't find a doc for this, but it seems the `x y` positions (`4 4`) are negative offsets from top left. So if we want to center the cursor to the center of the circle, we want to do `${cursorImageSizePx/2} ${cursorImageSizePx/2}`.